pkg/sql/plan2/function/function.go

```diff
@@ -0,0 +1,277 @@
+			Args: MakeLimitArgList([]Arg{
+				{Name: "left", Typ: types.T_uint8},
+				{Name: "right", Typ: types.T_uint8},
+			}),
+			ReturnTyp: types.T_uint8,
+			ID:        operatorEqualUint8Uint8,
+			Fn: func(vs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+				lv, rv := vs[0], vs[1]
+				lvs, rvs := lv.Col.([]uint8), rv.Col.([]uint8)
+				vec, err := process.Get(proc, int64(len(lvs)), lv.Typ)
+				if err != nil {
+					return nil, err
+				}
+				rs := encoding.DecodeUint8Slice(vec.Data)
+				rs = rs[:len(rvs)]
+				nulls.Or(lv.Nsp, rv.Nsp, vec.Nsp)
+				vector.SetCol(vec, add.Uint8Add(lvs, rvs, rs))
+				if lv.Ref == 0 {
+					process.Put(proc, lv)
+				}
+				if rv.Ref == 0 {
+					process.Put(proc, rv)
+				}
+				return vec, nil
+			},
+		},
+	},
+
+	"case": {
+		{
+			Name: "case_when_int64",
+			Flag: plan.Function_NONE,
+			Args: MakeUnLimitArgList(func(ts []types.T) bool {
+				l := len(ts)
+				if l < 3 {
+					return false
+				}
+
+				for i := 0; i < l-1; i += 2 { //case and then should be int64
+					if ts[i] != types.T_int64 && isNotNull(ts[i]) {
+						return false
+					}
+				}
+				if l%2 == 1 { // has else part
+					if ts[l-1] != types.T_int64 && isNotNull(ts[l-1]) {
+						return false
+					}
+				}
+				return true
+			}),
+			ReturnTyp: types.T_int64,
+			ID:        operatorCaseWhenInt64,
+			Fn: func(vs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+				return nil, nil
+			},
+		},
+	},
+	// Functions
+	// SubQuery
+}
+
+func MakeLimitArgList(args []Arg) ArgList {
+	return ArgList{
+		Limit:     true,
+		ArgTypes1: args,
+	}
+}
+
+func MakeUnLimitArgList(fn func([]types.T) bool) ArgList {
+	if fn == nil {
+		panic("function with variable-length parameters should have its own type-check function")
+	}
+
+	return ArgList{
+		Limit: false,
+		ArgTypes2: struct {
+			TypeCheckFn func(inputTypes []types.T) bool
+		}{TypeCheckFn: fn},
+	}
+}
+
+func AppendFunction(name string, newFunction Function) error {
+	if fs, ok := functionRegister[name]; ok {
+		for _, f := range fs {
+			if reflect.DeepEqual(f, newFunction) {
+				return errors.New(errno.DuplicateFunction, fmt.Sprintf("function %s(%v) existed.", name, f.Args))
+			}
+		}
+	}
+
+	registerMutex.Lock()
+	defer registerMutex.Unlock()
+
+	functionRegister[name] = append(functionRegister[name], newFunction)
+	return nil
+}
+
+// GetFunctionByID get function structure by its function id.
+func GetFunctionByID(name string, id int64) (Function, error) {
+	fs, ok := functionRegister[name]
+	if !ok {
+		return emptyFunction, errors.New(errno.UndefinedFunction, fmt.Sprintf("undefined function name '%s'", name))
+	}
+	for _, f := range fs {
+		if f.ID == id {
+			return f, nil
+		}
+	}
+	return emptyFunction, errors.New(errno.UndefinedFunction, fmt.Sprintf("undefined function id '%d'", id))
+}
+
+// GetFunctionByName check if a function exist or not by function name and arg types, if matches, return its function structure.
```

Comment:
I think you have done this backward -- You want to search function registration first, and for each function you find there, try to match args.